java-checks/src/main/java/org/sonar/java/checks/serialization/SerializableFieldInSerializableClassCheck.java

```diff
@@ -135,10 +173,14 @@ private static boolean isTransientSerializableOrInjected(VariableTree member) {
       return true;
     }
     SymbolMetadata metadata = member.symbol().metadata();
-    return metadata.isAnnotatedWith("javax.inject.Inject")
+    return metadata.isAnnotatedWith(JAVAX_INJECT)
       || metadata.isAnnotatedWith("javax.ejb.EJB")
       || metadata.isAnnotatedWith("org.apache.wicket.spring.injection.annot.SpringBean")
-      || metadata.annotations().stream().anyMatch(annotation -> annotation.symbol().isUnknown());
+      || hasAnnotationsWithIncompleteSemantic(metadata);
```

Comment:
`isAnnotatedWith` will go through all annotations, so with this code, you end up going through the list 4 times when the field is not annotated by any of the annotations (which I expect to be the most frequent case). We could do everything in one shot, without using `isAnnotatedWith`. 

The same happens in `isAnnotatedWithInject`.